docs/dependency_confusion.md

```diff
@@ -1,140 +1,55 @@
-> Npm packages follow the typical package file + lock file approach. The package file is
-> [package.json](https://docs.npmjs.com/cli/v6/configuring-npm/package-json), the lock file is
-> [package-lock.json](https://docs.npmjs.com/cli/v6/configuring-npm/package-lock-json) or
-> [npm-shrinkwrap.json](https://docs.npmjs.com/cli/v6/configuring-npm/shrinkwrap-json). Cachito
-> requires the lock file.
->
-> The lock file pins all dependencies to exact versions. For https dependencies, which are impossible
-> to pin, Cachito requires the integrity value (a checksum). For dependencies from the npm registry,
-> Cachito does not require integrity values, but newer versions of npm will always include them. Check
-> if all your dependencies have an integrity value, update your lock file if not.
->
-> Do not try to use private registries. If you point npm to a private registry (or any registry other
-> than the official one) via [.npmrc](https://docs.npmjs.com/cli/v6/configuring-npm/npmrc), Cachito
-> will ignore it and look in the official registry anyway.
->
-> ## yarn
->
-> *TL;DR: same as npm, but the handling of unofficial registries is slightly less surprising. Still,
-> you probably should not use them.*
->
-> Yarn packages are identical to npm packages except that they use
-> [yarn.lock](https://classic.yarnpkg.com/en/docs/yarn-lock/) as the lock file. Everything that
-> applies to npm applies to yarn.
->
-> The one difference is the handling of unofficial registries. If you point yarn to an unofficial
-> registry via .npmrc or [.yarnrc](https://classic.yarnpkg.com/en/docs/yarnrc), this will be reflected
-> in the resolved url in the lock file. Cachito will see that the url does not point to the official
-> registry and will treat the dependency as a plain https dependency. If the url is accessible to
-> Cachito, it will download the dependency directly without relying on npm/yarn dependency resolution.
-> That does not necessarily make using unofficial registries a good idea. If the registry is private,
-> your build will either fail or leak internal package names.
->
-> ## RubyGems (Bundler)
->
-> *TL;DR: allowing `https://rubygems.org` as the only source should mitigate the issue for GEM dependencies*
->
-> Cachito parses `Gemfile.lock` which pins all dependencies to exact versions. Cachito allows GEM dependencies
-> to be fetched only from `https://rubygems.org`, otherwise raises an error. GIT dependencies are specified using
-> a repo URL and pinned to a commit hash.
->
-> Bundler doesn't verify checksums of dependencies yet, however, there's an effort to bring
-> [internal GitLab implementation](https://gitlab.com/gitlab-org/gitlab/-/merge_requests/92633)
-> to [the upstream](https://github.com/rubygems/rubygems/pull/5808).
+### yarn
 
-[cachito-1]: https://github.com/containerbuildsystem/cachito
+- package manager for JavaScript
+- lock file: `yarn.lock`
+- [docs/yarn.md](https://github.com/containerbuildsystem/cachi2/tree/main/docs/npm.md)
```

Comment:
This is not an adequate replacement because you just replaced Yarn v1 content (which we don't support yet) with a reference to our Yarn v2+ docs (indended, **in reality you linked NPM instead**).